app/src/main/java/appricottsoftware/clarity/RegisterActivity.java

```diff
@@ -26,14 +51,115 @@ protected void onCreate(Bundle savedInstanceState) {
+                    surveyActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK|Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                    startActivity(surveyActivityIntent);
+                    finish();
+                } else {
+
+                    // After successful save of user info on back end
+                    // Switch to home activity
+
+
+                    Intent homeActivityIntent = new Intent(this, HomeActivity.class);
+                    homeActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                    startActivity(homeActivityIntent);
+                    finish();
+                }
+
+                ClarityApp.getRestClient().registerRequest(usernameString, emailString, hashedPassword, firstnameString, lastnameString, this, new JsonHttpResponseHandler() {
+                    @Override
+                    public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                        Log.e(TAG, "onSuccess1 : " + response.toString() );
+                        super.onSuccess(statusCode, headers, response);
+                    }
+
+                    @Override
+                    public void onSuccess(int statusCode, Header[] headers, JSONArray response) {
+                        Log.e(TAG, "onSuccess2 : " + response.toString());
+                        super.onSuccess(statusCode, headers, response);
+                    }
+
+                    @Override
+                    public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONObject errorResponse) {
+                        Log.e(TAG, "onFailue1 : " + errorResponse.toString());
+                        super.onFailure(statusCode, headers, throwable, errorResponse);
+                    }
+
+                    @Override
+                    public void onFailure(int statusCode, Header[] headers, Throwable throwable, JSONArray errorResponse) {
+                        Log.e(TAG, "onFailue2 : " + errorResponse.toString());
+                        super.onFailure(statusCode, headers, throwable, errorResponse);
+                    }
+
+                    @Override
+                    public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                        Log.e(TAG, "onFailue3 : " + responseString.toString());
+                        super.onFailure(statusCode, headers, responseString, throwable);
+                    }
+
+                    @Override
+                    public void onSuccess(int statusCode, Header[] headers, String responseString) {
+                        Log.e(TAG, "onSuccess3 : " + responseString.toString());
+                        super.onSuccess(statusCode, headers, responseString);
+                    }
+                });
+
                 break;
             default:
                 break;
         }
     }
-}
+
+    public String Hash_Password(String originalPassword) {
```

Comment:
Inconsistent style for method identifiers. We should probably change this to hashPassword( ... )

Reply:
Done